data_io/location.py

```diff
@@ -25,8 +26,18 @@ def __init__(self, *largs, **kwargs):
         super(URLKeyBuilder, self).__init__()
 
     def build(self, url: str) -> Location:
-
-        # Build the URL key
+        '''
+            default ports in builder
+        '''
+        # dictionary of defaul port
+        default_port = dict(oracle = '1521', redshift = '5439')
```

Comment:
Recommend changing the dictionary keys to the connection lookup keys (tuples).

Also, keyword args do not have spaces before/after assignment operator.

Example:

`oracle='1521'`